compiler/damlc/daml-ide/src/DA/Daml/LanguageServer/SplitGotoDefinition.hs

```diff
@@ -0,0 +1,239 @@
+nameSortExternalModule m | isExternalName m = nameModule_maybe m
+nameSortExternalModule _ = Nothing
+
+-- daml/tryGetDefinition :: TryGetDefinitionParams -> Maybe TryGetDefinitionResult
+tryGetDefinition :: IdeState -> TryGetDefinitionParams -> LSP.LspM c (Either ResponseError (Maybe TryGetDefinitionResult))
+tryGetDefinition ideState params = Right <$>
+  case uriToFilePath' $ tgdpTextDocument params ^. LSP.uri of
+    Nothing -> pure Nothing
+    Just (toNormalizedFilePath' -> file) ->
+      liftIO $ runActionSync ideState $ runMaybeT $ do
+        (loc, mName) <- MaybeT $ getDefinitionWithName file $ tgdpPosition params
+        let tgdName = do
+              name <- mName
+              m <- nameSortExternalModule name
+              pure $ TryGetDefinitionName
+                (moduleNameString $ moduleName m)
+                (unitIdString $ moduleUnitId m)
+                (occNameString $ nameOccName name)
+                (toTryGetDefinitionNameSpace $ occNameSpace $ nameOccName name)
+        pure $ TryGetDefinitionResult loc tgdName
+
+replaceChar :: Char -> Char -> String -> String
+replaceChar val replacement = fmap (\c -> if c == val then replacement else c)
+
+-- daml/gotoDefinitionByName :: GotoDefinitionByNameParams -> GotoDefinitionByNameResult
+gotoDefinitionByName :: IdeState -> GotoDefinitionByNameParams -> LSP.LspM c (Either ResponseError GotoDefinitionByNameResult)
+gotoDefinitionByName ideState params = do
+  mRoot <- LSP.getRootPath
+  liftIO $ runActionSync ideState $ exceptTToResult $ do
+    -- Working out the file by getting the IDE root and pSrc from daml.yaml, getting all the source files for it, then searching for our module
+    --   We search rather than explicitly building the path to account for pSrc being a daml file, whereby file discovery logic is via dependencies
+    -- I tried to do this better, by looking up the module name in IDE state, but it seems the IDE doesn't load
+    -- modules until you open the file, so it doesn't hold any context about "all" modules.
+    -- (trust me I tried so hard to make this work)
+    root <- hoistMaybe (Just "Failed to get IDE root") mRoot
+    projectConfig <- liftIO $ readProjectConfig (ProjectPath root)
+    config <- except $ first (Just . show) $ parseProjectConfig projectConfig
+
+    srcFiles <- maybeTToExceptT "Failed to get source files" $ getDamlFiles $ root </> pSrc config
+    -- Must be sorted shorted to longest, since we always want the shortest path that matches our suffix
+    -- to avoid accidentally picking Main.A.B.C if we're just looking for A.B.C
+    -- We also prefix all paths with "/" and search for our suffix starting with "/"
+    -- This is avoid incorrectly picking MA.B.C if we were looking for A.B.C and it didnt exist
+    let sortedSrcFiles = sortOn (Prelude.length . fromNormalizedFilePath) srcFiles
+        moduleSuffix = "/" <> replaceChar '.' '/' (tgdnModuleName $ gdnpName params) <> ".daml"
```

Comment:
does this work on windows?

Reply:
looking at this again, it's probably fine since we're going through `NormalizedFilePath`

Reply:
The whole multi-ide is optional, so I'm going to merge and test on windows explicitly once we have a release, since its quite painful to directly build on windows. (again this will not break windows IDE even if this logic is wrong)